src/alchemlyb/tests/parsing/test_namd.py

```diff
@@ -36,13 +48,183 @@ def test_u_nk_idws():
     assert u_nk.index.names == ['time', 'fep-lambda']
     assert u_nk.shape == (29252, 11)
 
-def test_u_nk_restarted():
+
+@pytest.fixture
+def restarted_dataset():
+    return load_restarted()
+
+
+def _corrupt_fepout(fepout_in, params):
+    """Corrups specific lines in a fepout file according to each line's prefix,
+    using caller-supplied functions.
+
+    Parameters
+    ----------
+    fepout_in: str
+        Path to fepout file to be modified. This file will not be overwritten.
+    
+    params: list of tuples
+        For each tuple, the first element must be a str that will be passed to
+        startswith() to identify the line(s) to modify (e.g. "#NEW"). The
+        second element must be a function that accepts a list of strs which is
+        the output of running split() on the identified line and returns
+        a modified list of tokens that will be reassembled into the line to be
+        output.
+
+    Returns
+    -------
+    The name of a temporary file which the caller must unlink.
+    """
+
+    fh, fepout_out = mkstemp(suffix='.fepout.bz2', prefix='restarted_')
+    close(fh)
+    with bz2.open(fepout_out, 'wt') as f_out:
+        with bz2.open(fepout_in, 'rt') as f_in:
+            for line in f_in:
+                for prefix, func in params:
+                    if line.startswith(prefix):
+                        line = ' '.join(func(line.split())) + '\n'
+                f_out.write(line)
+    return fepout_out
+
+
+@pytest.fixture
+def restarted_dataset_inconsistent():
+    """Returns intentionally messed up dataset."""
+    # We explicitly call load_restarted() again because we are modifying the
```

Comment:
You should be able to do this with normal function-scoped fixtures.

```python
@pytest.fixture
def restarted_dataset_inconsistent(restarted_dataset):
     filenames = restrarted_dataset['data']['both']
     ...
```
Here `restarted_dataset` will be a unique copy.